### ISSUE TITLE
Add Snowplow build context

### ISSUE THREAD

**Issue by WillDaSilva** [meltano/meltano#6171]
Currently we include `is_dev_build` in the Snowplow environment context, which generally indicates whether the events being fired is from a build of Meltano taken from PyPI, or a build built directly off of the Meltano repository.

We could extend this idea creating a Snowplow build context. When Meltano is built, a custom build tool which wraps `poetry.core.masonry.api` (as described in #3401) could collect info about the build. The data collected could include:

- The date and time the build was performed
- The timezone the build was performed in
- An equivalent to `is_dev_build` from the environment context
- Something like `is_ci_environment` from the environment context, but now for whether the build occurred in CI, as opposed to the execution of Meltano
- Whether `pipx` was used (by inspecting the process tree)
- More info about the build tools used and their versions
- The git commit built off of
- Whether the git repository was clean when the build was performed
- etc.

The custom build tool would collect all of the desired information, and store it in a JSON file in the wheel or sdist it builds. Then when events are fired by Meltano, this info could be included in a context just like how we include the environment context.

@aaronsteers @pnadolny13 @pandemicsyn

**Comment by pnadolny13** [comment]
@WillDaSilva I'm not super in tune with the build process or the things we care about in the process but what are the use cases for this data? What behaviors are we trying to understand with this context?

My initial thought is that from the data/reporting side of things the build context isnt all that interesting but I might just might not understand the use cases. Wouldnt most of these built contexts be the same since the majority of users are pip installing? Are we trying to see if users are building their own package?

**Comment by WillDaSilva** [comment]
> Are we trying to see if users are building their own package?

@pnadolny13 Yes. In particular we want to know if the package was built using `pipx`. As [@aaronsteers noted](https://github.com/meltano/meltano/pull/5978#discussion_r889606266), the `is_dev_build` value from the environment context does not properly account for this.

This context would also be helpful in determining if users are building Meltano for themselves more generally, and if they're doing so with modifications.

**Comment by WillDaSilva** [comment]
@pnadolny13 @aaronsteers I think it is time to revisit this idea. The current plan for exposing Meltano Cloud alpha/beta CLI features is to do so in a feature branch, and have the users install Meltano using `pip install git+https://github.com/meltano/meltano@cloud` (or some other branch/tag instead of `cloud`). The telemetry will consider these to be dev builds, which I believe will cause their data to be filtered out after we collect it.

To avoid that, we can introduce the build context as proposed above. It will contain information gathered within `scripts/build.py`, which is run when the wheel is built. With this we can solve the above problem by including the git branch/tag found when `scripts/build.py` runs. I've tested this out, and it works as expected - pip clones the repository, then builds the wheel within it, so we can run `git branch --show-current` in addition to collecting other information that might help us down the line.

Then when we're processing the collected data we can avoid filtering out data based on `is_dev_build` if that build was done on the `cloud` branch.

**Comment by pnadolny13** [comment]
@WillDaSilva for context - as of today we dont filter based on the `is_dev_build` attribute. That's something we could look into doing if it would accurately filter out devs vs real users. It felt too risky to me originally to use it for any low level filtering because I dont know all the ways people can install the package so wouldnt want to accidentally over filter with an attribute I dont fully understand.

**Comment by WillDaSilva** [comment]
@pnadolny13 Thanks for letting me know. That makes this unnecessary in light of the recent build changes, and the proposed Meltano Cloud alpha/beta installation method, which is convenient. I'll keep this issue open in case we ever decide that we're interested in gathering this information, but for now I'll unassigned myself.
